src/splitpane/SplitPane.ts

```diff
@@ -0,0 +1,190 @@
+
+const DEFAULT_SIZE = 100;
+
+@theme(css)
+export default class SplitPane extends SplitPaneBase<SplitPaneProperties> {
+	private _divider: HTMLElement;
+	private _dragging: boolean;
+	private _lastSize: number;
+	private _position: number;
+	private _root: HTMLElement;
+	private _boundHandlers: any[];
+
+	constructor() {
+		/* istanbul ignore next: disregard transpiled `super`'s "else" block */
+		super();
+
+		/**
+		 * `mouseup` and other events aren't triggered when a user's cursor leaves div.root, so
+		 * global handlers are need to listen to the document instead. SplitPane
+		 * uses a `_dragging` flag so no handlers will be erroneously executed
+		 * if a user isn't actually resizing this SplitPane instance.
+		 */
+		this._boundHandlers = [];
+		[
+			{ event: 'mouseup', func: this._onDragEnd.bind(this) },
+			{ event: 'mousemove', func: this._onDragMove.bind(this) },
+			{ event: 'touchmove', func: this._onDragMove.bind(this) }
+		].forEach(object => {
+			document.addEventListener(object.event, object.func);
+			this._boundHandlers.push(object);
+		});
+
+		this.own(createHandle(() => {
+			this._boundHandlers.forEach(object => document.removeEventListener(object.event, object.func));
+		}));
+
+		const viewportSubscription = observeViewport({
+			next: debounce(this._onResize.bind(this), 200)
+		});
+
+		this.own(createHandle(() => {
+			viewportSubscription.unsubscribe();
+		}));
+	}
+
+	private _getPosition(event: MouseEvent & TouchEvent) {
+		const { direction = Direction.row } = this.properties;
+
+		if (direction === Direction.row) {
+			return event.changedTouches ? event.changedTouches[0].clientX : event.clientX;
+		}
+		else {
+			return event.changedTouches ? event.changedTouches[0].clientY : event.clientY;
+		}
+	}
+
+	private _onDragStart(event: MouseEvent & TouchEvent) {
+		this._dragging = true;
+		this._position = this._getPosition(event);
+	}
+
+	private _onDragMove(event: MouseEvent & TouchEvent) {
+		if (!this._dragging) {
+			return;
+		}
+
+		const {
+			direction = Direction.row,
+			onResize,
+			size = DEFAULT_SIZE
+		} = this.properties;
+
+		const currentPosition = this._getPosition(event);
+		let newSize = (this._lastSize === undefined ? size : this._lastSize) + currentPosition - this._position;
+
+		const maxSize = direction === Direction.row ?
+			this._root.offsetWidth - this._divider.offsetWidth :
+			this._root.offsetHeight - this._divider.offsetHeight;
+
+		this._lastSize = newSize;
+
+		newSize = newSize < 0 ? 0 : newSize;
+		newSize = newSize > maxSize ? maxSize : newSize;
+
+		this._position = currentPosition;
+
+		onResize && onResize(newSize);
+	}
+
+	private _onDragEnd(event: MouseEvent & TouchEvent) {
+		this._dragging = false;
+	}
+
+	private _onResize() {
+		const {
+			direction = Direction.row,
+			onResize,
+			size = DEFAULT_SIZE
+		} = this.properties;
+
+		const rootSize = direction === Direction.row ? this._root.offsetWidth : this._root.offsetHeight;
```

Comment:
@smhigley I actually switched back to using `flex: 0 0 auto` for the leading div and `flex: 1 1 0%` for the trailing div. This is the more intuitive user experience to me, and how I originally coded it: the user can size the SplitPane, the end. The SplitPane doesn't attempt any cleverness during a window resize that may or may not cause the pane to resize. Things got messy when multiple levels of nested SplitPanes had incorrect explicit widths / heights set after window resizing since flexbox only uses these values as suggestions in certain cases.